### PR TITLE
[FLINK-11583][configuration] Support deprecated and fallback keys at once

### PR DESCRIPTION
**Pull request by zentol**
## What is the purpose of the change

Modify `ConfigOption#withFallbackKeys` and `ConfigOption#withDeprecatedKeys` to take deprecated/fallback keys that were configured previously also into account.

## Brief change log

* merge newly passed keys with existing keys in both methods
* setup ordering such that fallback keys are always at the front, and deprecated keys always at the end

## Verifying this change

see `ConfigurationTest#testFallbackAndDeprecatedKeys`

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/configuration/ConfigOption.java [hunk 1]
```diff
@@ -134,10 +138,13 @@
 	 * @return A new config options, with the given deprecated keys.
 	 */
 	public ConfigOption<T> withDeprecatedKeys(String... deprecatedKeys) {
-		FallbackKey[] fallbackKeys = Arrays.stream(deprecatedKeys)
-			.map(FallbackKey::createDeprecatedKey)
+		final Stream<FallbackKey> newFallbackKeys = Arrays.stream(deprecatedKeys).map(FallbackKey::createFallbackKey);
```
Comment (tillrohrmann) [inline flink-core/src/main/java/org/apache/flink/configuration/ConfigOption.java]: Should be `FallbackKey::createDeprecatedKey`

Comment (tillrohrmann) [reply flink-core/src/main/java/org/apache/flink/configuration/ConfigOption.java]: maybe rename variable to `newDeprecatedKeys`

Comment (zentol) [reply flink-core/src/main/java/org/apache/flink/configuration/ConfigOption.java]: I'll fix the method reference.

The naming scheme is messed up generally in this context, both deprecated and fallback keys are stored as `FallbackKey` in a `fallbackKeys` field :/

I think I'll rename this one to `newDeprecatedKey`, and the union of keys to `mergedAlternativeKeys` or something...

Comment (zentol) [reply flink-core/src/main/java/org/apache/flink/configuration/ConfigOption.java]: I'll also check if I can't add a test that deprecated keys are indeed considered deprecated.

Comment (tillrohrmann) [reply flink-core/src/main/java/org/apache/flink/configuration/ConfigOption.java]: Good idea. Thanks Chesnay.

**Comment by zentol** [comment]
@tillrohrmann Do you want to take another look?

File: flink-core/src/test/java/org/apache/flink/configuration/ConfigOptionTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests for the {@link ConfigOption}.
+ */
+public class ConfigOptionTest {
```
Comment (tillrohrmann) [inline flink-core/src/test/java/org/apache/flink/configuration/ConfigOptionTest.java]: Let's extend `TestLogger` here

**Pull request merged**